misc/showvte.1
.TH SHOWVTE 1
.SH NAME
showvte \- show build-time configuration of evilvte
.SH SYNOPSIS
.B showvte
.RI [ option ]
.SH DESCRIPTION
.B showvte
displays build-time configuration of evilvte.
.SH OPTIONS
.TP
.B \-v
Show version of showvte.
.SH SEE ALSO
.BR evilvte (1)
.SH AUTHOR
showvte was written by Wen-Yen Chuang <caleb AT calno DOT com>


ChangeLog
UNRELEASED
 * Make close button smaller
 * Improve resizing behavior with "-VteTerminal-inner-border" property

0.5.1~pre1

 * Temporary workaround for "-VteTerminal-inner-border" property
 * Mouse clicking can also turn off urgency hint now

0.5.0

 The "One binary to rule them all!" Release

 * Fix a bug for -e option
 * Fix a bug for input method menu
 * New configure option: --with-gtk=2+3|3+2
   The binary will not link to VTE nor GTK+, thus can support GTK+ 2.x and 3.x
   at the same time. It still depends on VTE runtime, of course.
   Use --2 or --3 command line options to choose the version of GTK+.
   If evilvte segfaults with --3 option, please try export USE_ACCESSIBLE=0 or
   USE_ACCESSIBLE=1.
 * Drop codes with VTE version < 0.17.1

0.5.0~pre1
 * Sync COLOR_STYLE: ZENBURN with xterm
   Thanks to Artur Dryomov <me AT ming13 DOT ru>
   Previous ZENBURN is renamed to ZENBURN_DARK
 * Replace all deprecated functions that have suitable replacements
 * Make everything GTK+ 3.1.92 compatible
   Use GTK_FONT_CHOOSER_DIALOG only for GTK+ version >= 3.1.90
 * Make everything VTE 0.29.1 compatible
   If you encounter problems, try export CFLAGS="-DUSE_ACCESSIBLE=0"
   and rebuild evilvte.

0.4.9
 * Fix makefile parallelization
   Thanks to Thomas Weißschuh <thomas_weissschuh AT lavabit DOT com>
 * Use GTK_FONT_CHOOSER_DIALOG for GTK+ version >= 3.1.12
 * Expand VTE widget in GtkGrid

0.4.9~pre5
 * Support GtkGrid
 * Fix an SCROLLBAR minor bug with option OFF_L and OFF_R
 * New icon set under the license:
   GNU General Public License version 2 or any later version
   Thanks to Alexander Inglessi <inglessi AT gmail DOT com>
 * Use vte_terminal_fork_command() by default. Please see:
   http://evilvte.blogspot.com/2011/03/issue-evilvte-048-makes-zsh-segfault.html

0.4.9~pre4
 * Support SEARCH_CASE_SENSITIVE
   I will make it toggleable at runtime in the future.
   But I am lazy to write it now. :-P
   Patches are very welcome.
 * Make everything GLib 2.29.8 compatible
 * Sign SHA1SUMS using RSA key ID BE8BA819 and DSA key ID E98D6159

0.4.9~pre3
 * Grayed out menu items according to their status:
   "Copy", "Paste", "Zoom default"

0.4.9~pre2
 * Minor tweaks in TABBAR_AUTOHIDE for GTK+ 3.x
 * Hide VTE_FIXED color style which was merged in VTE version >= 0.17.1
 * Drop !TAB codes since evilvte is GtkNotebook-only for a very long time.
   For those who do not use tabs, the binary size will increase a little.

0.4.9~pre1
 * Support "Auto-click" when there is only one effective menu item.
   You can assign anything to right-click now.
 * Support embedded GTK3_CSS

0.4.8

 The "Welcome, GTK+ 3.0.0!" Release

 * Better compatibility with string matching
 * Fix a bug that made MATCH_STRING fail when MENU is not enabled
   Thanks to Kabakov Boris <pigritor AT gmail DOT com>
 * New COLOR_STYLE option: ZENBURN
   Thanks to thagraal and <ezrgzx AT gmail DOT com>
 * New options: MATCH_STRING_HTTP
                MATCH_STRING_MAIL
                MATCH_STRING_FILE
 * Hide MATCH_STRING
 * Make everything GTK+ 3.0.0 compatible

0.4.8~pre3
 * Fix a bug for multiple -e options in command line
 * Fix an i18n bug when linked to GTK+ version >= 2.90.0

0.4.8~pre2
 * Initial support for vte_terminal_search_*
   The feature of VTE is still quite buggy.
   Please see: https://bugzilla.gnome.org/627886
 * Support underscore in LABEL_* to indicate mnemonic of menu item
 * Drop MATCH_STRING_EXEC which was deprecated since 0.4.7~pre1
 * Hide DEFAULT_EMULATION_TYPE, TAB_BORDER_HORIZONTAL and TAB_BORDER_VERTICAL
 * Provide a clearsigned SHA1SUMS using DSA key ID E98D6159

0.4.8~pre1
 * Status bar and scrollbar can be hidden initially and be toggled on later
 * Better behavior for *_TOGGLE_TABBAR with TABBAR_AUTOHIDE enabled
 * Sync window state with window manager
 * Fix some cross-platform issues
   Thanks to Aragon Gouveia <aragon AT phat DOT za DOT net>
 * Initial support for new resizing behavior of VTE
   If you are using GTK+ version >= 2.91.1 and VTE version >= 0.27.1,
   the window size of evilvte is not always correct currently.

0.4.7.1
 * Fix a segfault bug of HOTKEY_MIMIC_SCROLL_UP and HOTKEY_MIMIC_SCROLL_DOWN

0.4.7
 * Better compatibility with WM_CLASS
   - Accept environment variable RESOURCE_NAME
   - Document --name and --class command line options in manpage
   - Note that -title and -T no longer changes WM_CLASS
   - Thanks to Zveroy <zveroy AT gmail DOT com> and
               Tetralet <tetralet AT gmail DOT com>
 * Accept environment variable PKG_CONFIG_PATH, thanks to letoh
 * Support -g +X+Y command line option to assign initial window geometry
 * Prevent tab labels getting focus

0.4.7~pre5
 * Support -fn command line option to assign font and font size
 * Better compatibility with configure options

0.4.7~pre4
 * Support HOTKEY_MIMIC_SCROLL_*, thanks to qotsa
 * New configure option: --with-gtk=2.0|3.0

0.4.7~pre3
 * Support vte_terminal_fork_command_full
 * Support all positive integer font size
 * Drop gdk-pixbuf dependency. It was used for a bug only exists in VTE 0.24.0.

0.4.7~pre2
 * Support BELL_URGENT, aka WM_URGENT hint. Thanks to L29Ah.

0.4.7~pre1

 The "Thanks to Aragon!" Release

 * Rename MATCH_STRING_EXEC to MATCH_STRING_L
 * Provide MATCH_STRING_M
 * Fix two MATCH_STRING_* bugs
 * Fix two WINDOW_TITLE_DYNAMIC bugs
 * Fix a cross-platform issue

0.4.6
 * Workaround for VTE 0.24.0 vte_terminal_set_background_image_file() bug

0.4.6~pre6
 * Support CURSOR_SHAPE
 * New BACKSPACE_KEY and DELETE_KEY option: ERASE_TTY
 * Simplify Makefile and configure

0.4.6~pre5
 * Saturation can be adjusted by holding Ctrl and scrolling mouse up or down.
   Define both MOUSE_CTRL_SATURATION and HOTKEY to TRUE to use this function.
   If CTRL_ALT is TRUE, you have to hold Ctrl and Alt while scrolling.
 * Remove close button border in some specific themes
   Thanks to Tetralet <tetralet AT gmail DOT com>

0.4.6~pre4
 * Support MATCH_STRING_EXEC

src/showvte.sh
#!/bin/sh
# Copyright (C) 2008-2009  Wen-Yen Chuang <caleb AT calno DOT com>
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; version 2 of the License.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License along
# with this program; if not, write to the Free Software Foundation, Inc.,
# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.

  head -n 1 $0 > src/showvte
  tail -n 6 $0 | grep -v ^grep >> src/showvte
  sed 's/\t/ /g' src/custom.h | tr -s ' ' ' ' | sed -e 's/^ //' -e 's~/\*~\n~g' | grep ^\#define >> src/showvte

SHOWVTE_PROG_NAME=`grep PROGRAM_NAME src/custom.h | tr -s ' ' ' ' | sed 's/^ //' | grep -v ^\/\/ | awk '{print $3}' | sed 's/"//g'`
if [ "$SHOWVTE_PROG_NAME" = "" ]; then
  SHOWVTE_PROG_NAME="evilvte"
fi

SHOWVTE_VERSION=`grep PROGRAM_VERSION src/custom.h | grep -v ^\/\/ | awk '{print $3}'`
if [ "$SHOWVTE_VERSION" = "" ]; then
  SHOWVTE_VERSION=`head -n 1 Changelog`
fi

echo echo Configuration of $SHOWVTE_PROG_NAME $SHOWVTE_VERSION: >> src/showvte

DEFINE_LINES=`grep ^\#define src/showvte | wc | awk '{print $1}'`

if [ "$DEFINE_LINES" = "0" ]; then
  echo echo [1m[31mDisable everything.[m >> src/showvte
  echo exit >> src/showvte
fi

  grep ^grep $0 >> src/showvte
  chmod 755 src/showvte
  exit

if [ "$1" = "-v" ]; then
  SHOWVTE_VERSION=`grep ^echo $0 | cut -d : -f 1 | cut -d ' ' -f 5`
  echo showvte, version $SHOWVTE_VERSION
  exit
fi
grep ^#define $0 | sed -e 's/$/[m/' -e 's/"/[1m[32m"/' -e 's/ 0/[1m[32m 0/' -e 's/ 1/[1m[32m 1/' -e 's/ 2/[1m[32m 2/' -e 's/ 3/[1m[32m 3/' -e 's/ 4/[1m[32m 4/' -e 's/ 5/[1m[32m 5/' -e 's/ 6/[1m[32m 6/' -e 's/ 7/[1m[32m 7/' -e 's/ 8/[1m[32m 8/' -e 's/ 9/[1m[32m 9/' -e 's/TRUE/[1m[32mTRUE/' -e 's/FALSE/[1m[31mFALSE/' -e 's/ AUTO/[1m[32m AUTO/' -e 's/ BACKSPACE/[1m[32m BACKSPACE/g' -e 's/BACKSPACE_KEY/[mBACKSPACE_KEY/' -e 's/ DELETE/[1m[32m DELETE/g' -e 's/DELETE_KEY/[mDELETE_KEY/' -e 's/ VTE_FIXED/[1m[32m VTE_FIXED/' -e 's/ LINUX/[1m[32m LINUX/' -e 's/ RXVT/[1m[32m RXVT/' -e 's/ TANGO/[1m[32m TANGO/' -e 's/ XTERM/[1m[32m XTERM/' -e 's/ g_/[1m[32m g_/' -e 's/ RIGHT/[1m[32m RIGHT/' -e 's/ LEFT/[1m[32m LEFT/' -e 's/ TOP/[1m[32m TOP/' -e 's/ BOTTOM/[1m[32m BOTTOM/' -e 's/ CTRL /[1m[32m CTRL /'
